tanner/api.py

```diff
@@ -38,3 +100,52 @@ def __init__(self):
             for (i, val) in enumerate(query_res):
                 query_res[i] = json.loads(val)
         return query_res
+
+    async def return_session_info(self, sess_uuid, snare_uuid= None):
+        query_res = []
+        if snare_uuid:
+            snare_uuids = [snare_uuid]
+        else:
+            snare_uuids = await self.return_snares()
+
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            if sessions == 'Invalid SNARE UUID':
+                continue
+            for sess in sessions:
+                if sess['sess_uuid'] == sess_uuid:
+                    return sess
+
+    async def return_sessions(self, filters):
+        query_res = []
+        if 'snare_uuid' in filters:
+            snare_uuids = [filters['snare_uuid']]
+        else:
+            snare_uuids = await self.return_snares()
+
+        matching_sessions = []
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            for sess in sessions:
+                if sessions == 'Invalid SNARE UUID':
+                    continue
+                is_matching_sesssion = True
+                if 'user-agent' in filters:
+                    if filters['user-agent'] not in sess['user-agent']:
+                        is_matching_sesssion = False
+                if 'peer_ip' in filters:
+                    if filters['peer_ip'] != sess['peer_ip']:
+                        is_matching_sesssion = False
+                if 'attack_type' in filters:
+                    if filters['attack_type'] not in sess['attack_types']:
+                        is_matching_sesssion = False
+                if 'time_interval' in filters:
+                    if filters['time_interval']['end_time'] < sess['start_time'] or filters['time_interval']['start_time'] > sess['end_time']:
+                        is_matching_sesssion = False
+                if 'possible_owner' in filters:
+                    if filters['possible_owner'] not in sess['possible_owners']:
+                        is_matching_sesssion = False
+
+                if is_matching_sesssion:
```

Comment:
default value is True, so for every non-valid filter string it returns snare-uuids (see comment with screenshot)